pkg/services/datasources/service/datasource.go

```diff
@@ -243,6 +252,21 @@ func (s *Service) UpdateDataSource(ctx context.Context, cmd *datasources.UpdateD
 			return err
 		}
 
+		if cmd.Name != "" && cmd.Name != dataSource.Name {
+			query := &datasources.GetDataSourceQuery{
+				Name:  cmd.Name,
+				OrgID: cmd.OrgID,
+			}
```

Comment:
does this work if the ID is not specified?

Reply:
Yes it does, see https://github.com/grafana/grafana/blob/9ebede1e18dd8c3a2630e05a37d613b1c5f0ab36/pkg/api/datasources.go#L590-L593

Reply:
OrgID is must to get datasource and the unique constraint is applied on `org_id` + `name`
https://github.com/grafana/grafana/blob/cca9d897339ae2de087bc5b7177849a730d4b38d/pkg/services/sqlstore/migrations/datasource_mig.go#L68